server/auth/auth.go

```diff
@@ -748,13 +748,13 @@ func (auth *AuthManager) handleConnect(conn comms.Link, msg *msgjson.Message) *m
 	acctInfo, paid, open := auth.storage.Account(user)
 	if acctInfo == nil {
 		return &msgjson.Error{
-			Code:    msgjson.AuthenticationError,
-			Message: "no account info found for account ID" + connect.AccountID.String(),
+			Code:    msgjson.AccountNotFoundError,
+			Message: "no account found for account ID: " + connect.AccountID.String(),
 		}
 	}
 	if !paid {
 		return &msgjson.Error{
-			Code:    msgjson.AuthenticationError,
+			Code:    msgjson.UnpaidAccountError,
```

Comment:
I realize that this error isn't being explicitly checked anymore in production code (it is in tests), but I still think this is a good change.

Reply:
Leaving it there until I hear otherwise.

Reply:
More specific is good. A brief search seems to indicate that nobody is relying on `AuthenticationError`. Even though this is a breaking change that would normally require a release version bump, we're allowing that right now, so this is fine.